src/Common/src/Interop/Windows/Winsock/Interop.WinsockAsync.cs

```diff
@@ -108,9 +108,13 @@ internal struct TransmitPacketsElement
             [System.Runtime.InteropServices.FieldOffset(8)]
             internal Int64 fileOffset;
             [System.Runtime.InteropServices.FieldOffset(8)]
-            internal IntPtr buffer;
+            internal byte* buffer;
             [System.Runtime.InteropServices.FieldOffset(16)]
             internal IntPtr fileHandle;
+
+            // This controls how many TransmitPacketsElement structs we will allocate on the stack.
+            // Beyond this, we need to alloc on the heap and pin.
+            internal const int StackAllocLimit = 128;
```

Comment:
This is potentially 3K on the stack.  I'm wondering if that's too much, and whether the limit should be more like 42 instead of 128 for a 1K limit.